dry_rest_permissions/generics.py

```diff
@@ -120,14 +124,35 @@ def has_permission(self, request, view):
             if hasattr(model_class, action_method_name):
                 return getattr(model_class, action_method_name)(request)
 
-        if request.method in permissions.SAFE_METHODS:
-            assert hasattr(model_class, 'has_read_permission'), \
-                self._get_error_message(model_class, 'has_read_permission', action_method_name)
-            return model_class.has_read_permission(request)
-        else:
-            assert hasattr(model_class, 'has_write_permission'), \
-                self._get_error_message(model_class, 'has_write_permission', action_method_name)
-            return model_class.has_write_permission(request)
+        try:
+            if request.method in permissions.SAFE_METHODS:
+                assert hasattr(model_class, 'has_read_permission'), \
+                    self._get_error_message(model_class, 'has_read_permission', action_method_name)
+                return model_class.has_read_permission(request)
+            else:
+                assert hasattr(model_class, 'has_write_permission'), \
+                    self._get_error_message(model_class, 'has_write_permission', action_method_name)
+                return model_class.has_write_permission(request)
+        except AssertionError as e:
+            if self.django_permissions:
+                django_permission, django_permission_exists = None, False
+                if hasattr(view, 'action'):
+                    action = self._get_action(view.action)
+                    django_actions_map = {'list': 'view', 'retrieve': 'view', 'update': 'change',
+                                          'destroy': 'delete', 'create': 'add'}
+                    django_action = django_actions_map[action] if action in django_actions_map else action
+                    django_permission, django_permission_exists = self._get_permission(model_class, django_action)
+                django_generic_action = f'view' if request.method in permissions.SAFE_METHODS else f'change'
+                django_generic_permission, dgp_exists = self._get_permission(model_class, django_generic_action)
```

Comment:
gdp_exists should be django_generic_permission_exists to be consistent with django_permission_exists, nothing blocking here just my OCD